### ISSUE TITLE
Custom resource uses wrong datatype for version

### ISSUE THREAD

**Issue by mklenbw** [serverless/serverless#12632]
### Issue description

Serverless creates a custom resource for the cloudwatch-role. The version-property is set as number instead of a string. This let's linters like cfn-lint fail.

To solve this just put quotes around it.

lib/plugins/aws/package/compile/events/lib/ensure-api-gateway-cloud-watch-role.js
````
cfTemplate.Resources[customResourceLogicalId] = {
      Type: 'Custom::ApiGatewayAccountRole',
      Version: 1.0,    // <- this should be '1.0'
      Properties: {
        ServiceToken: {
          'Fn::GetAtt': [customResourceFunctionLogicalId, 'Arn'],
        },
        RoleArn: configuredRoleArn,
      },
    }
````

### Context

N/A

**Comment by eahefnawy** [comment]
Thanks for the heads up @mklenbw ... I pushed a fix for this in https://github.com/serverless/serverless/pull/12647 and will be included in the next release 🚀

**Comment by mklenbw** [comment]
Will this be fixed in v3 as well?

**Comment by Mmarzex** [comment]
@mklenbw as we are only doing security fixes for v3 now there is no plan at this time for change like this to be backported.

**Comment by Mmarzex** [comment]
This has been released with [4.1.7](https://github.com/serverless/serverless/releases/tag/v4.1.7)
